File: backends/aspnet/src/Controllers/PlazaController.cs
using HelloTimePro.Aspnet.Services;
using HelloTimePro.Aspnet.Web;
using Microsoft.AspNetCore.Mvc;

namespace HelloTimePro.Aspnet.Controllers;

[Route("api/v1/plaza/capsules")]
public sealed class PlazaController : ApiControllerBase
{
    private readonly AuthContext _authContext;
    private readonly PlazaService _plaza;
    private readonly CapsuleService _capsules;

    public PlazaController(AuthContext authContext, PlazaService plaza, CapsuleService capsules)
    {
        _authContext = authContext;
        _plaza = plaza;
        _capsules = capsules;
    }

    [HttpGet]
    public async Task<IActionResult> List()
    {
        var viewer = await _authContext.Optional(AuthHeader);
        var sort = StringParam("sort") ?? "new";
        var filter = StringParam("filter") ?? "all";
        var q = StringParam("q");
        return Wrapped(await _plaza.PlazaList(sort, filter, q, Page, PageSize, viewer?.Id));
    }

    [HttpGet("{id}")]
    public async Task<IActionResult> Detail(string id)
    {
        var viewer = await _authContext.Optional(AuthHeader);
        return Wrapped(await _capsules.GetPlazaDetail(id, viewer?.Id));
    }
}

File: fullstacks/rails/app/controllers/api/v1/plaza_controller.rb
module Api
  module V1
    class PlazaController < BaseController
      def index
        viewer = optional_user(auth_header)
        result = PlazaService.plaza_list(
          sort: request.query_parameters["sort"] || "new",
          filter: request.query_parameters["filter"] || "all",
          q: request.query_parameters["q"],
          page: page_param, page_size: page_size_param, viewer_id: viewer&.id,
        )
        render_ok(result)
      end

      def show
        viewer = optional_user(auth_header)
        render_ok(CapsuleService.get_plaza_detail(params[:id], viewer&.id))
      end
    end
  end
end
